WordPress/src/main/java/org/wordpress/android/ui/stories/StoryMediaSaveUploadBridge.kt

```diff
@@ -0,0 +1,170 @@
+                    freshlyTaken = false, // we don't care about this
+                    editorMediaListener = this@StoryMediaSaveUploadBridge,
+                    doUploadAfterAdding = true
+            )
+            postUtils.preparePostForPublish(editPostRepository.getEditablePost()!!, site)
+            savePostToDbUseCase.savePostToDb(WordPress.getContext(), editPostRepository, site)
+
+            if (networkUtils.isNetworkAvailable()) {
+                postUtils.trackSavePostAnalytics(
+                        editPostRepository.getPost(),
+                        site
+                )
+                uploadService.uploadPost(WordPress.getContext(), editPostRepository.id, true)
+                // SAVED_ONLINE
+            } else {
+                // SAVED_LOCALLY
+            }
+        }
+    }
+    // endregion
+
+    private fun cancelAddMediaToEditorActions() {
+        job.cancel()
+    }
+
+    @Subscribe(sticky = true, threadMode = MAIN)
+    fun onEventMainThread(event: StorySaveResult) {
+        // only trigger the bridge preparation and the UploadService if the Story is now complete
+        // otherwise we can be receiving successful retry events for individual frames we shoulnd't care about just
+        // yet.
+        if (isStorySuccessfullySavedAndComplete(event)) {
+            // only remove it if it was successful - we want to keep it and show a snackbar once when the user
+            // comes back to the app if it wasn't, see MySiteFrament for details.
+            EventBus.getDefault().removeStickyEvent(event)
+            event.metadata?.let {
+                val site = it.getSerializable(WordPress.SITE) as SiteModel
+                editPostRepository.loadPostByLocalPostId(it.getInt(StoryComposerActivity.KEY_POST_LOCAL_ID))
+                addNewStoryFrameMediaItemsToPostAndUploadAsync(site, event)
+                // TODO WPSTORIES add TRACKS
+                // lets add an EVENT for START UPLOADING MEDIA
+                // AnalyticsTracker.track(Stat.STORIES_BLA_BLA_ADDED_MEDIA_OR_SOMETHING);
+            }
+        } else {
+            // TODO WPSTORIES add TRACKS for ERROR
+            // AnalyticsTracker.track(Stat.MY_SITE_ICON_UPLOAD_UNSUCCESSFUL);
+        }
+    }
+
+    private fun isStorySuccessfullySavedAndComplete(event: StorySaveResult): Boolean {
+        return (event.isSuccess() &&
+                event.frameSaveResult.size == StoryRepository.getStoryAtIndex(event.storyIndex).frames.size)
+    }
+
+    override fun appendMediaFiles(mediaFiles: Map<String, MediaFile>) {
+        // Create a gallery shortcode and placeholders for Media Ids
+        val idsString = mediaFiles.map {
+            WP_STORIES_POST_MEDIA_LOCAL_ID_PLACEHOLDER + it.value.id.toString()
+        }.joinToString(separator = ",")
+        editPostRepository.update { postModel: PostModel ->
+            postModel.setContent("[gallery type=\"slideshow\" ids=\"$idsString\"]")
+            true
+        }
+    }
```

Comment:
This logic isn't clear to eye at a glance so I would want us to encapsulate this in a use case and write some tests to verify it's behavior. If so we could create an issue to track this. Wdyt?

Reply:
Gotcha, thanks for this comment. Even when we're going to remove this part and change it for an actual Gutenberg block (Tug is working on that), it makes sense to encapsulate the "block creation" in a use case as suggested. Would this be okay to do on a separate PR after alpha1 when we focus on refactors / testability or do you think we should do it here? Let me know 🙏 🙇

Reply:
> Would this be okay to do on a separate PR after alpha1 when we focus on refactors / testability or do you think we should do it here? Let me know 🙏 🙇

Honestly, we can do this after alpha when we focus on the reactors. Once we are tracking these things I am good 😄

Reply:
Tracked in the Stories repository project 👍